atlasdb-timelock-server/src/test/java/com/palantir/atlasdb/timelock/atomix/AtomixTimestampServiceIntegrationTest.java

```diff
@@ -60,44 +61,57 @@ public static void stopAtomix() {
     @Before
     public void setupTimestampService() {
         DistributedLong distributedLong = DistributedValues.getTimestampForClient(ATOMIX_REPLICA, CLIENT_KEY);
-        timestampService = new AtomixTimestampService(distributedLong);
+        atomixTimestampService = new AtomixTimestampService(distributedLong);
     }
 
     @Test
-    public void timestampsShouldBeIncreasing() {
-        long ts1 = timestampService.getFreshTimestamp();
-        long ts2 = timestampService.getFreshTimestamp();
-
-        assertThat(ts1).isLessThan(ts2);
+    public void timestampsAreReturnedInOrder() {
+        TimestampServiceTests.timestampsAreReturnedInOrder(atomixTimestampService);
     }
 
     @Test
     public void canRequestTimestampRange() {
-        int expectedNumTimestamps = 5;
-        TimestampRange range = timestampService.getFreshTimestamps(expectedNumTimestamps);
-
-        long actualNumTimestamps = range.getUpperBound() - range.getLowerBound() + 1;
-        assertThat(actualNumTimestamps)
-                .withFailMessage("Expected %d timestamps, got %d timestamps. (The returned range was: %d-%d)",
-                        expectedNumTimestamps, actualNumTimestamps, range.getLowerBound(), range.getUpperBound())
-                .isEqualTo(expectedNumTimestamps);
+        TimestampServiceTests.canRequestTimestampRangeWithGetFreshTimestamps(atomixTimestampService);
+    }
+
+    @Test
+    public void timestampRangesAreReturnedInNonOverlappingOrder() {
+        TimestampServiceTests.timestampRangesAreReturnedInNonOverlappingOrder(atomixTimestampService);
+    }
+
+    @Test
+    public void willNotHandOutTimestampsEarlierThanAFastForward() {
+        TimestampServiceTests.willNotHandOutTimestampsEarlierThanAFastForward(
+                atomixTimestampService,
+                atomixTimestampService);
+    }
+
+    @Test public void
+    willDoNothingWhenFastForwardToEarlierTimestamp() {
+        TimestampServiceTests.willDoNothingWhenFastForwardToEarlierTimestamp(
+                atomixTimestampService,
+                atomixTimestampService);
+    }
+
+    @Ignore
+    @Test
+    public void canReturnManyUniqueTimestampsInParallel() throws TimeoutException, InterruptedException {
```

Comment:
If you want to ignore this test just for Atomix, I believe you can override the test method and `@Ignore` it.